.github/workflows/build-and-test.yml

```diff
@@ -121,13 +125,14 @@ jobs:
       run: |
         source env/activate
         set -o pipefail # Ensures that the exit code reflects the first command that fails
-        pytest --junit-xml=${{ steps.strings.outputs.test_report_path }} 2>&1 | tee pytest.log
+        pip install pytest-split
```

Comment:
We can move "pytest-split" to requirements.txt in the future, I didn't want to do it in this change as it will trigger Docker image rebuild.